src/main/steps/tasklistresponse/proceedings/ProceedingDetailsMapper.ts

```diff
@@ -0,0 +1,172 @@
+              value: otherDetailsInfo,
+            };
+            proceedingDetailsList.push(proceedingDetailsInfo);
+          } else {
+            const otherDetailsInfo: OtherProceedingDetails = {
+              orderDetail: nestedOrder.orderDetail,
+              caseNo: nestedOrder.caseNo,
+              currentOrder: nestedOrder.currentOrder,
+              orderCopy: nestedOrder.orderCopy,
+              orderDate: getLocalDate(nestedOrder.orderDate),
+              orderEndDate: getLocalDate(nestedOrder.orderEndDate),
+            };
+            const proceedingDetailsInfo: ProceedingsOrderDataInterface = {
+              id: '',
+              value: otherDetailsInfo,
+            };
+            proceedingDetailsList.push(proceedingDetailsInfo);
+          }
+        });
+        const proceedings: Proceedings = {
+          orderType: `${order}`,
+          proceedingDetails: proceedingDetailsList,
+        };
+        const proceedingData: ProceedingDetailsData = {
+          id: '',
+          value: proceedings,
+        };
+        proceedingDetails.push(proceedingData);
+      }
+      currentOrPreviousProceedings.haveChildrenBeenInvolvedInCourtCase = req.session.userCase.proceedingsStart;
+      currentOrPreviousProceedings.courtOrderMadeForProtection = req.session.userCase.proceedingsStartOrder;
+      currentOrPreviousProceedings.proceedingsList = proceedingDetails;
+    });
+    respondent.value.response.currentOrPreviousProceedings = currentOrPreviousProceedings;
+  } else {
+    respondent.value.response.currentOrPreviousProceedings = currentOrPreviousProceedings;
+  }
+
+  return respondent;
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const getProceedingDetails = (respondent: Respondent, req: AppRequest): Partial<CaseWithId> => {
+  const currentProceedings = respondent.value.response.currentOrPreviousProceedings;
+  const courtProceedingsOrders: ProceedingsOrderTypes[] = [];
+  const proceedingOrderTypeInterface: ProceedingsOrderTypeInterface = {};
+  if (currentProceedings) {
+    req.session.userCase.proceedingsStart = currentProceedings.haveChildrenBeenInvolvedInCourtCase;
+    req.session.userCase.proceedingsStartOrder = currentProceedings.courtOrderMadeForProtection;
+    const proceedingList = currentProceedings.proceedingsList;
+    proceedingList?.forEach(proceedings => {
+      const proceedingOrderInterfaceList: ProceedingsOrderInterface[] = [];
+      const orderType = proceedings.value.orderType as ProceedingsOrderTypes;
+      courtProceedingsOrders.push(orderType);
+      req.session.userCase.courtProceedingsOrders = courtProceedingsOrders;
+      let id = 1;
+      proceedings.value.proceedingDetails?.forEach(proceeding => {
+        console.log(proceeding.value.caseNo);
+        console.log(getDisplayDate(proceeding.value.orderDate));
+
+        const proceedingOrderInterface: ProceedingsOrderInterface = {
+          id: getNextId(id),
```

Comment:
The name of the function and the actual purpose is not in sync. Probably you can. move increment of the id inside the function itself.

Reply:
Made changes as suggested